src/main.cpp

```diff
@@ -2614,7 +2618,10 @@ void txRxProcessing()
 
             // zero speech input just in case infifo2 underflows
             memset(insound_card, 0, nsam_in_48*sizeof(short));
-            codec2_fifo_read(cbData->infifo2, insound_card, nsam_in_48);
+            int nread = codec2_fifo_read(cbData->infifo2, insound_card, nsam_in_48);
+
+            // special case - if we are ending Tx then only process if we have samples available from mic
+            if (endingTx && (nread != nsam_in_48)) break;
```

Comment:
codec2_fifo_read() seems to return -1 if there isn't enough data and 0 otherwise. Could that be why significant chunks of audio still seem to be getting cut?

Reply:
Good catch @tmiw - I should check the return codes on stuff I wrote 10 years ago :grinning: